### ISSUE TITLE
在推理时发现和onnx的结果对不上，然后发现NCNN在全连接层时发生错误

### ISSUE THREAD

**Issue by busishengui** [Tencent/ncnn#4546]
## error log | 日志或报错信息 | ログ
输出全为0或者NAN
## context | 编译/运行环境 | バックグラウンド
Window/Linux都一样
## how to reproduce | 复现步骤 | 再現方法
这是我的param
7767517
49 49
Input            input_x:0                0 1 input_x:0
MemoryData       const_fold_opt__88       0 1 const_fold_opt__88 0=1
MemoryData       fc1/biases/read/_61__cf__61:0 0 1 fc1/biases/read/_61__cf__61:0 0=64
MemoryData       fc1/weights/read/_62__cf__62:0 0 1 fc1/weights/read/_62__cf__62:0 0=64 1=960
MemoryData       fc2/biases/read/_63__cf__63:0 0 1 fc2/biases/read/_63__cf__63:0 0=64
MemoryData       fc2/weights/read/_64__cf__64:0 0 1 fc2/weights/read/_64__cf__64:0 0=64 1=64
MemoryData       logits/logits_b/read/_65__cf__65:0 0 1 logits/logits_b/read/_65__cf__65:0 0=2
MemoryData       logits/logits_w/read/_66__cf__66:0 0 1 logits/logits_w/read/_66__cf__66:0 0=2 1=64
Reshape          Reshape                  1 1 input_x:0 Reshape:0 0=40 1=150
ExpandDims       ExpandDims               2 1 Reshape:0 const_fold_opt__88 ExpandDims:0 -23303=0
Reshape          conv1/conv1__13          1 1 ExpandDims:0 conv1/conv1__13:0 0=40 1=150 2=1
Convolution      Conv__74                 1 1 conv1/conv1__13:0 conv1/batch_normalization/FusedBatchNorm:0 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=72
ReLU             Relu                     1 1 conv1/batch_normalization/FusedBatchNorm:0 Relu:0
Pooling          MaxPool                  1 1 Relu:0 MaxPool:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__75                 1 1 MaxPool:0 conv2/batch_normalization/FusedBatchNorm:0 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=576
ReLU             Relu_1                   1 1 conv2/batch_normalization/FusedBatchNorm:0 Relu_1:0
Pooling          MaxPool_1                1 1 Relu_1:0 MaxPool_1:0 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__76                 1 1 MaxPool_1:0 conv3/batch_normalization/FusedBatchNorm:0 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=1152
ReLU             Relu_2                   1 1 conv3/batch_normalization/FusedBatchNorm:0 Relu_2:0
Pooling          MaxPool_2                1 1 Relu_2:0 MaxPool_2:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__77                 1 1 MaxPool_2:0 conv4/batch_normalization/FusedBatchNorm:0 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=2304
ReLU             Relu_3                   1 1 conv4/batch_normalization/FusedBatchNorm:0 Relu_3:0
Pooling          MaxPool_3                1 1 Relu_3:0 MaxPool_3:0 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=1 5=1
Convolution      Conv__78                 1 1 MaxPool_3:0 conv5/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=4608
ReLU             Relu_4                   1 1 conv5/batch_normalization/FusedBatchNorm:0 Relu_4:0
Pooling          MaxPool_4                1 1 Relu_4:0 MaxPool_4:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__79                 1 1 MaxPool_4:0 conv6/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=9216
ReLU             Relu_5                   1 1 conv6/batch_normalization/FusedBatchNorm:0 Relu_5:0
Pooling          MaxPool_5                1 1 Relu_5:0 MaxPool_5:0 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__80                 1 1 MaxPool_5:0 conv7/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=9216
ReLU             Relu_6                   1 1 conv7/batch_normalization/FusedBatchNorm:0 Relu_6:0
Pooling          MaxPool_6                1 1 Relu_6:0 MaxPool_6:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__81                 1 1 MaxPool_6:0 conv8/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=9216
ReLU             Relu_7                   1 1 conv8/batch_normalization/FusedBatchNorm:0 Relu_7:0
Pooling          MaxPool_7                1 1 Relu_7:0 MaxPool_7:0 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=1 15=1 5=1
Convolution      Conv__82                 1 1 MaxPool_7:0 conv9/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=9216
ReLU             Relu_8                   1 1 conv9/batch_normalization/FusedBatchNorm:0 Relu_8:0
Pooling          MaxPool_8                1 1 Relu_8:0 MaxPool_8:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Convolution      Conv__83                 1 1 MaxPool_8:0 conv10/batch_normalization/FusedBatchNorm:0 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=9216
ReLU             Relu_9                   1 1 conv10/batch_normalization/FusedBatchNorm:0 Relu_9:0
Pooling          MaxPool_9                1 1 Relu_9:0 MaxPool_9:0 0=0 1=1 11=1 2=1 12=1 3=0 13=0 14=0 15=0 5=1
Permute          MaxPool_9__72            1 1 MaxPool_9:0 MaxPool_9__72:0 0=3
Reshape          Flatten/flatten/Reshape  1 1 MaxPool_9__72:0 Flatten/flatten/Reshape:0 0=-1
Gemm             fc1/fc1/MatMul_Gemm__6   3 1 Flatten/flatten/Reshape:0 fc1/weights/read/_62__cf__62:0 fc1/biases/read/_61__cf__61:0 fc1/fc1/MatMul_Gemm__6:0 0=1.000000e+00 1=1.000000e+00 2=0 3=0
ReLU             Relu_10                  1 1 fc1/fc1/MatMul_Gemm__6:0 Relu_10:0
Gemm             fc2/fc2/MatMul_Gemm__7   3 1 Relu_10:0 fc2/weights/read/_64__cf__64:0 fc2/biases/read/_63__cf__63:0 fc2/fc2/MatMul_Gemm__7:0 0=1.000000e+00 1=1.000000e+00 2=0 3=0
ReLU             Relu_11                  1 1 fc2/fc2/MatMul_Gemm__7:0 Relu_11:0
Gemm             logits/logits/MatMul_Gemm__8 3 1 Relu_11:0 logits/logits_w/read/_66__cf__66:0 logits/logits_b/read/_65__cf__65:0 logits/logits/MatMul_Gemm__8:0 0=1.000000e+00 1=1.000000e+00 2=0 3=0
Softmax          labels_softmax           1 1 logits/logits/MatMul_Gemm__8:0 labels_softmax:0 0=0 1=1
输入一个1x6000的向量，
在Flatten/flatten/Reshape 层我发现输出的shape还是正常的，是960的一维向量，
然后经过fc1/fc1/MatMul_Gemm__6，就莫名其妙地变成了64*16的输出向量，这里应该是64的一维向量活或者64*1的一维向量才对，所以从这里开始后面的推理跟着全部错误

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
